linter.lua

```diff
@@ -25,10 +26,53 @@ local function run_lint_cmd(path, linter)
 end
 
 
+local function match_pattern(text, pattern, order)
+  if order == nil then
+    return text:gmatch(pattern)
+  end
+
+  return coroutine.wrap(function()
+    for one, two, three in text:gmatch(pattern) do
+      local line, col, warn
+      for field,position in pairs(order) do
+        local value
```

Comment:
Instead of a switch to get the data, a table could be used.

```lua
local fields = {one, two, three}
local value = fields[position] or nil
```

Reply:
the fields table can be created outside of the inner loop so it is only created once for each gmatch